pkg/cmd/codespace/ssh.go

```diff
@@ -526,27 +571,36 @@ func (a *App) printOpenSSHConfig(ctx context.Context, opts sshOptions) (err erro
 			result := sshResult{}
 			defer wg.Done()
 
-			session, err := codespaces.ConnectToLiveshare(ctx, a, noopLogger(), a.apiClient, cs)
+			codespaceConnection, err := codespaces.GetCodespaceConnection(ctx, a, a.apiClient, cs)
 			if err != nil {
 				result.err = fmt.Errorf("error connecting to codespace: %w", err)
-			} else {
-				defer safeClose(session, &err)
-
-				invoker, err := rpc.CreateInvoker(ctx, session)
-				if err != nil {
-					result.err = fmt.Errorf("error connecting to codespace: %w", err)
-				} else {
-					defer safeClose(invoker, &err)
-
-					_, result.user, err = invoker.StartSSHServer(ctx)
-					if err != nil {
-						result.err = fmt.Errorf("error getting ssh server details: %w", err)
-					} else {
-						result.codespace = cs
-					}
-				}
+				sshUsers <- result
+				return
+			}
+
+			fwd, err := portforwarder.NewPortForwarder(ctx, codespaceConnection)
```

Comment:
Do we need a `defer fwd.Close()`? I noticed that we did that in the `stdio` branch. If so, we also need to add it to other commands

Reply:
We do it in the [`ForwardPortToListener`](https://github.com/dmgardiner25/cli/blob/7dd70c2161a6f9e11abd3aa3127db4a72ec707fb/internal/codespaces/portforwarder/port_forwarder.go#L70) func so that most of the calls to forward a port don't need to worry about it. The `stdio` branch has to do it because it's not calling `ForwardPortToListener`. Admittedly, just calling it `Close` isn't very descriptive because it's closing the SSH connection on the tunnel client so I've updated it to `CloseSSHConnection` to make that more clear.